firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java

```diff
@@ -202,6 +203,15 @@ void runTransaction(String action, Runnable operation) {
     return value;
   }
 
+  public static void clearPersistence(
+      Context context, DatabaseId databaseId, String persistenceKey) {
+    String databaseName = SQLitePersistence.databaseName(persistenceKey, databaseId);
+    String sqlLitePath = context.getDatabasePath(databaseName).getPath();
+    String journalPath = sqlLitePath + "-journal";
+    new File(sqlLitePath).delete();
```

Comment:
Previously these had warnings about ignoring a return value. Now that this is production code we probably shouldn't ignore anymore. Specifically if we run into permission problems or something where the file still exists after trying to delete it, this call should throw.

Use `java.nio.file.Path#delete()` instead and only fail here if the error is not file not found.

Reply:
Actually, whoops! `java.nio.file.Path` Was added in API 26 and we need to support back to API 16. We still need to use the old way, but if there’s an error check to see if the file still exists. If still exists then throw

Reply:
Actually, whoops! `java.nio.file.Path` Was added in API 26 and we need to support back to API 16. We still need to use the old way, but if there’s an error check to see if the file still exists. If still exists then throw